core/base/src/TError.cxx

```diff
@@ -90,7 +91,7 @@ static void DebugPrint(const char *fmt, ...)
    va_end(ap);
 
    // Serialize the actual printing.
-   R__LOCKGUARD2(gErrorMutex);
+   std::lock_guard<std::mutex> guard(gErrorMutex);
```

Comment:
We may want to note somewhere that this is a change of behavior.  Before the lock was taken only if ROOT's thread support was turned on.  Now it is always taken.

Reply:
I guess not ...